niizarr/_compat.py

```
@@ -0,0 +1,137 @@
+        import numcodecs
+        compressor_map = {
+            "blosc": numcodecs.Blosc,
+            "zlib": numcodecs.Zstd,
+        }
+    elif zarr_version == 3:
+        import zarr.codecs
+        compressor_map = {
+            "blosc": zarr.codecs.BloscCodec,
+            "zlib": zarr.codecs.ZstdCodec,
+        }
+    else:
+        raise ValueError(f"zarr version {zarr_version} is not supported")
+    if name not in compressor_map:
+        raise ValueError('Unknown compressor', name)
+    Compressor = compressor_map[name]
+
+    return Compressor(**kwargs)
+
+
+def _swap_header(header):
+    """
+    Byteswap the given header array based on the installed numpy version.
+
+    For numpy versions < 2.0.0, it uses header.newbyteorder().
+    For numpy versions >= 2.0.0, it uses header.view(header.dtype.newbyteorder()).
+
+    Parameters:
+        header (np.ndarray): The numpy array (or structured array) whose byte order needs to be swapped.
+
+    Returns:
+        np.ndarray: The array with its byte order swapped.
+    
+    Note: newbyteorder() does not change data in memory, it only changes how data is interpreted.
+          byteswap() changes the data in memory.
+    """
+    if NumpyVersion(np.__version__) < NumpyVersion("2.0.0"):
+        return header.newbyteorder()
+    else:
+        return header.view(header.dtype.newbyteorder())
+
+
+def _open_zarr_group(out,
+                     mode: Literal["r", "w"] = "w",
+                     store_opt: Optional[dict] = None,
+                     **kwargs):
+    store_opt = store_opt or {}
+    if pyzarr_version == 3:
+        StoreLike = zarr.storage.StoreLike
+        FsspecStore = zarr.storage.FsspecStore
+        LocalStore = zarr.storage.LocalStore
+    else:
+        StoreLike = zarr.storage.Store
+        FsspecStore = zarr.storage.FSStore
+        LocalStore = zarr.storage.DirectoryStore
+        if "zarr_version" in kwargs:
+            if kwargs["zarr_version"] != pyzarr_version:
```

Should we raise an error if the requested zarr version is 3?

---

yes we can do that.